data/common.yaml

```diff
@@ -20,6 +23,13 @@ mcollective::server_config:
   plugin.yaml: "/etc/puppetlabs/mcollective/generated-facts.yaml"
   registerinterval: 10
   registration: "Choria"
+  logfile: "/var/log/puppetlabs/mcollective.log"
```

Comment:
I do not question the need for the parameters, I do question the location (init class) however.

> for example in the official mysql module see override_options.

They only have one parameter, well named. It is clear these options become are merged last. We have three parameters: `server`, `client`, `common`. We should drop `common` and rename the other two as as server/client overrides.

> `$options = mysql_deepmerge($mysql::params::default_options, $override_options)`

Maybe we should rather go with this also, making default options parameters *outside* the init class to make it clear where we get our defaults from.

Question remains, what do we do if a user sets up the proposed parameter `logfile = /var/log/server.log` *and* `common_config = {logfile = /client.log}`?